### ISSUE TITLE
False negative `too-few-format-args`

### ISSUE THREAD

**Issue by clavedeluna** [PyCQA/pylint#8012]
### Bug description

```
"{1}".format(20)
"{2}".format(20)
```

both should emit `too-few-format-args` because the interpreter will complain with `IndexError: Replacement index N out of range for positional args tuple"

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
Formatting a regular string which could be a f-string (consider-using-f-string)
```

### Expected behavior

Too few arguments for format string (too-few-format-args)

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.13.0-dev0
Python 3.8.9 (default, Apr 13 2022, 08:48:07)
```

**Comment by clavedeluna** [comment]
Actually, given examples of `bad-format-string`, maybe this is supposed to emit that message instead?
